src/main/java/edu/kit/kastel/sdq/intelligrade/extensions/settings/ArtemisSettings.java

```diff
@@ -87,7 +87,22 @@ public class ArtemisSettings implements Configurable {
             this.apply();
             PluginState.getInstance().connect();
         });
-        contentPanel.add(loginButton, "span 2, growx");
+        contentPanel.add(loginButton, "span 1, growx");
+
+        // Button to log out of artemis
+        var logoutButton = new JButton("Logout");
+        logoutButton.addActionListener(a -> {
+            // request a logout
+            boolean loggedOut = PluginState.getInstance().logout();
+            // clear data iff logout was actually carried out
+            if (loggedOut) {
+                ArtemisCredentialsProvider.getInstance().setJwt(null);
+                ArtemisCredentialsProvider.getInstance().setArtemisPassword(null);
+                ArtemisSettingsState.getInstance().setJwtExpiry(null);
```

Comment:
I'd prefer this to be part of PluginState#logout

Reply:
done